File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let minimumPrice= Number.MAX_VALUE;
    let maximumProfit= 0;
    for (let price of prices){
        if(price< minimumPrice) minimumPrice=price;
        else if((price-minimumPrice)>maximumProfit) maximumProfit=price-minimumPrice;
    }
    return maximumProfit;
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int buy = Integer.MAX_VALUE;
        int maxi = 0;
        for(int price:prices){
            if(price<buy) buy= price;
            else maxi = Math.max(maxi, price-buy);
        }
        return maxi;
    }
}